examples/website/i3s-building-scene-layer/src/components/building-explorer.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import styled from 'styled-components';
+import {Font, Color} from './styles';
+import {Checkbox, CheckboxOption, CheckboxSpan} from './checkbox';
+import {faAngleDown, faAngleRight, faCircle} from '@fortawesome/free-solid-svg-icons';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {IconProp, SizeProp} from '@fortawesome/fontawesome-svg-core';
+import {Sublayer} from '../helpers/sublayers';
+import {DropDown} from './drop-down';
+
+const BuildingExplorerContainer = styled.div`
+  max-height: 100%;
+  width: 246px;
+  overflow: auto;
+  align-items: flex-start;
+  position: absolute;
+  display: flex;
+  flex-direction: column;
+  background: #0e111a;
+  border-radius: 8px;
+  padding-top: 8px;
+`;
+
+const BuildingExplorerSublayers = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const CollapseContainer = styled.div`
+  margin-right: 5px;
+  cursor: pointer;
+`;
+
+const CheckboxContainer = styled.div`
+  ${Color}
+  ${Font}
+  margin-left: 10px;
+`;
+
+const Label = styled.h3`
+  margin: 0 16px 8px 16px;
+  cursor: pointer;
+  color: white;
+  font-weight: normal;
+`;
+
+const TopLabel = styled.h2`
+  margin: 0 16px 8px 16px;
+  cursor: pointer;
+  color: white;
+  font-weight: normal;
+`;
+
+type BuildingExplorerProps = {
+  isShown: boolean;
+  sublayers: Sublayer[];
+  buildingLevels: (string | number)[];
+  onSelect: (item: string) => void;
+  onUpdateSublayerVisibility: (sublayer: Sublayer) => void;
+};
+
+export const BuildingExplorer = ({
+  isShown,
+  sublayers,
+  buildingLevels,
+  onSelect,
+  onUpdateSublayerVisibility
+}: BuildingExplorerProps) => {
+  const toggleGroup = (sublayer: Sublayer) => {
+    sublayer.expanded = !sublayer.expanded;
+    onUpdateSublayerVisibility(sublayer);
+  };
+
+  const toggleSublayer = (sublayer: Sublayer) => {
+    sublayer.visibility = !sublayer.visibility;
+    onUpdateSublayerVisibility(sublayer);
+    setChildren(sublayer.sublayers, sublayer.visibility);
+  };
+
+  const setChild = (sublayer: Sublayer, isShown: boolean) => {
+    sublayer.visibility = isShown;
+    onUpdateSublayerVisibility(sublayer);
+    setChildren(sublayer.sublayers, isShown);
+  };
+
+  const setChildren = (sublayers: Sublayer[] | undefined, isShown: boolean) => {
+    if (sublayers) {
+      for (const sublayer of sublayers) {
+        setChild(sublayer, isShown);
+      }
+    }
+  };
+
+  const renderSublayers = (sublayers: Sublayer[]) => {
```

Comment:
Just as my preference, I prefer `function` syntax, at least for longer functions, so that is how most examples in this repo will look. But we don't necessarily need to be consistent

Reply:
Done